File: app/src/main/java/app/grapheneos/pdfviewer/loader/DocumentPropertiesAsyncTaskLoader.java
package app.grapheneos.pdfviewer.loader;

import android.content.Context;
import android.net.Uri;

import androidx.annotation.Nullable;
import androidx.loader.content.AsyncTaskLoader;

public class DocumentPropertiesAsyncTaskLoader extends AsyncTaskLoader<DocumentPropertiesResult> {

    public static final String TAG = "DocumentPropertiesLoader";

    public static final int ID = 1;

    private final String properties;
    private final int numPages;
    private final Uri uri;

    public DocumentPropertiesAsyncTaskLoader(Context context, String properties, int numPages, Uri uri) {
        super(context);

        this.properties = properties;
        this.numPages = numPages;
        this.uri = uri;
    }


    @Override
    protected void onStartLoading() {
        forceLoad();
    }

    @Nullable
    @Override
    public DocumentPropertiesResult loadInBackground() {

        DocumentPropertiesLoader loader = new DocumentPropertiesLoader(
                getContext(),
                properties,
                numPages,
                uri
        );

        return loader.loadAsResult();
    }
}

File: app/src/main/java/app/grapheneos/pdfviewer/loader/DocumentPropertiesAsyncTaskLoader.kt
package app.grapheneos.pdfviewer.loader

import android.content.Context
import android.net.Uri
import androidx.loader.content.AsyncTaskLoader
import androidx.loader.content.Loader

class DocumentPropertiesAsyncTaskLoader(context: Context, private val mProperties: String, private val mNumPages: Int, private val mUri: Uri):
    AsyncTaskLoader<List<CharSequence>>(context) {

    companion object {
        const val TAG = "DocumentPropertiesLoader"
        const val ID = 1
    }

    //Workaround till PdfViewer is migrated to Kotlin
    fun asLoader() = this as Loader<List<CharSequence>>

    override fun onStartLoading() {
        super.onStartLoading()
        forceLoad()
    }

    override fun loadInBackground(): List<CharSequence> {
        val loader = DocumentPropertiesLoader(
            context,
            mProperties,
            mNumPages,
            mUri
        )
        return loader.loadAsList()
    }
}
